ADRs/RH/SIG-SRE/ADR-00002 Open Source Documentation.md

```diff
@@ -0,0 +1,93 @@
+This document proposes a solution for maintaining an upstream version of the SOPs and having  a mechanism for consuming them downstream, as well as vice versa.
+
+## Goals
+
+* Propose a model for maintaining 2 versions of documents, for upstream and downstream consumption
+* Provide a worked example of this model as a reference
+
+## Non-goals
+
+* Implement a complete solution for a product or service
+
+## Current Architecture
+
+A Standard Operating Procedure (SOP) is a document used by SREs in response to alerts and other events.
+An example of what a current SOP architecture looks like is the Red Hat OpenShift Streams for Apache Kafka (RHOSAK) managed service. The RHOSAK SOPs are located in a private repository, as the SOPs contain references to Red Hat specific internal information that is not suitable for consumption by external users. Such information includes links to Vault secrets, information about how RHOSAK specifically set up their stage and production environments and information on the infrastructure itself, for example what certain custom resources should look like. The flow of how information is contributed to the RHOSAK SOPs is for an engineer or SRE to submit PRs to the repo.
+
+## Proposed Architecture
+
+The proposed solution is to follow a two repository model. One private repository that will contain SOPs that have the Red Hat internal information mentioned above, the other being a public repository that will have SOPs with the Red Hat internal information removed. This model will allow the SRE teams to use the private SOP repository without any information loss and allow upstream users to use and contribute to the SOP without breaching any Red Hat data policies.
+
+![Upstream, Downstream Pattern](images/upstream_downstream_docs_pattern.png)
+
+The content of the SOPs will be mirrored/synced between both repositories apart from the Red Hat internal information. The flow of contributing information to the repositories can be done both ways i.e from downstream to upstream and from upstream to downstream. This information can be contributed manually through the creation of pull requests or it can be done autonomously using the mirroring features in the repository. The team incharge of the SOPs will be responsible for maintaining the information between both repositories. This includes merging contributions from external users in the upstream repository and ensuring, if applicable, it gets added to the downstream repository and vice versa.
```

Comment:
```suggestion
The content of the SOPs will be mirrored/synced between both repositories apart from the Red Hat internal information. The flow of contributing information to the repositories can be done both ways i.e. from downstream to upstream and from upstream to downstream. This information can be contributed manually through the creation of pull requests or it can be done autonomously using the mirroring features in the repository. The team in charge of the SOPs will be responsible for maintaining the information between both repositories. This includes merging contributions from external users in the upstream repository and ensuring, if applicable, it gets added to the downstream repository and vice versa.
```